test/unit/api/term_black.h

```diff
@@ -24,14 +24,520 @@ class TermBlack : public CxxTest::TestSuite
   void setUp() override {}
   void tearDown() override {}
 
-  void testTermAssignment()
-  {
-    Term t1 = d_solver.mkReal(1);
-    Term t2 = t1;
-    t2 = d_solver.mkReal(2);
-    TS_ASSERT_EQUALS(t1, d_solver.mkReal(1));
-  }
+  void testEq();
+  void testGetKind();
+  void testGetSort();
+  void testIsNull();
+  void testNotTerm();
+  void testAndTerm();
+  void testOrTerm();
+  void testXorTerm();
+  void testEqTerm();
+  void testImpTerm();
+  void testIteTerm();
+
+  void testTermAssignment();
 
  private:
   Solver d_solver;
 };
+
+void TermBlack::testEq()
+{
+  Sort uSort = d_solver.mkUninterpretedSort("u");
+  Term x = d_solver.mkVar("x", uSort);
+  Term y = d_solver.mkVar("y", uSort);
+  Term z;
+
+  TS_ASSERT_THROWS_NOTHING(x == x);
+  TS_ASSERT(x == x);
```

Comment:
Suggestion: use `TS_ASSERT_EQUALS(x, x)` instead (it has some nicer output if something goes wrong).

Reply:
We decided offline to not use `TS_ASSERT_EQUALS` and `TS_ASSERT_DIFFERS` in this case since they can't be used for `!(x != x)` and the like. So for consistency, we use `TS_ASSERT` here.